src/ipaperftest/core/plugin.py

```diff
@@ -177,7 +177,8 @@ def generate_metadata(self, ctx):
             )
 
         for c in self.generate_clients(ctx):
-            machine_configs.append(c)
+            if c:
```

Comment:
What's the reason for this?

Reply:
IIRC I was running with zero clients, none are necessary. But something was being returned and my being impatient I think this was my solution rather than having the generator return nothing. I'll take another look at it.

Reply:
Right, so generate_clients() by default returns None so there is no machine_config to append, hence it blows up later in the code. yielding () instead has basically the same effect.